exercises/practice/rational-numbers/.meta/proof.ci.wren

```diff
@@ -0,0 +1,70 @@
+class Rational {
+  construct new(numerator, denominator) {
+    var gcd = Math.gcd(numerator.abs, denominator.abs)
+    var m = denominator < 0 ? -1 : 1
+    _num = m * numerator / gcd
+    _den = m * denominator / gcd
+  }
+
+  numerator { _num }
+  denominator { _den }
+
+  toString { "%(numerator)/%(denominator)" }
+
+  ==(other) {
+    assertRational(other)
+    return numerator == other.numerator &&
+           denominator == other.denominator
+  }
+
+  negated    { type.new(-numerator, denominator) }
+  reciprocal { type.new(denominator, numerator) }
+  abs        { type.new(numerator.abs, denominator) }
+
+  assertRational(other) {
+    if (!(other is this.type)) {
+      Fiber.assert("Not a %(type): %(other)")
```

Comment:
Should this be `this.type`?  I haven't written Wren for a bit... and I don't see any tests for this... if we aren't going to have any tests should this validation even be in the proof?

Reply:
Or perhaps this.type should be type?

Reply:
I don't recall why I even added that (it was months ago). Stripping it out.